autoload/tagbar.vim

```diff
@@ -2286,6 +2286,12 @@ function! s:JumpToTag(stay_in_tagbar) abort
     " Mark current position so it can be jumped back to
     mark '
 
+    " Check if the tag is already visible in the window.  We must do this
+    " before jumping to the line.
+    let topline = line('w0')
+    let bottomline = line('w$')
+    let alreadyvisible = (taginfo.fields.line >= topline) && (taginfo.fields.line <= bottomline)
```

Comment:
This logic looks fine, but I would like to request te abstract it to a new function that just returns a boolean answer for whether a line is visible. I suspect in the future we'll get a real API for this and it would be easier to swap out (and also use other places in our code) if it was abstracted as a function.

Additionally this would mean it could be called in the `if` statement *after* the other check, hence we can avoid the overhead of running this check if we're not going to use the answer anyway.

Reply:
Sure I can put it into a function.  However, I believe that we cannot call it in the `if` statement because, by the time we get there, the cursor has already been moved by the line:

```
execute taginfo.fields.line
```

This I believe will jump to the tag line, and in doing so, will change the values of the `topline` and `bottomline`, hence we need to call the function before this line.  Does that make sense?

Reply:
If you like, for optimization purposes, I can just avoid calling the function unless the option is on, e.g.:

```vim
let noscroll = 0
if g:tagbar_jump_lazy_scroll != 0
  let noscroll = call s:is_line_already_visible(taginfo.fields.line)
endif
```

then later down in the `if` statement we only need to check `if noscroll != 0`.

Reply:
On review I think you may be right about the logic here (although I didn't try to code it up the other way to proove the case). Let's go ahead with at least the small Optimization of checking whether the options is set before we do the extra math. With that done as you suggested here I think this will be ready for merge.

Reply:
Done